plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java

```diff
@@ -1681,6 +1687,290 @@ public void testSupportedNonPartitionedColumnMappingNoneWrites()
+        }
+    }
+
+    @Test
+    public void testCreateOrReplaceTableAsOnCdcEnabledTables()
+    {
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "test_create_or_replace_with_cdv", " (a BIGINT) WITH (change_data_feed_enabled = true)")) {
+            assertQueryFails(
+                    "CREATE OR REPLACE TABLE " + table.getName() + " AS SELECT 1 new_column",
+                    "CREATE OR REPLACE is not supported for tables with change data feed enabled");
+        }
+    }
+
+    @Test
+    public void testCreateOrReplaceTableWithSameLocationForManagedTable()
+    {
+        try (TestTable table = new TestTable(
+                getQueryRunner()::execute,
+                "create_or_replace_with_same_location_",
+                " (a BIGINT)")) {
+            HiveMetastore metastore = TestingDeltaLakeUtils.getConnectorService(getDistributedQueryRunner(), HiveMetastoreFactory.class)
+                    .createMetastore(Optional.empty());
+            String location = metastore.getTable("test_schema", table.getName()).orElseThrow().getStorage().getLocation();
+            assertTableType("test_schema", table.getName(), MANAGED_TABLE.name());
+            assertUpdate("CREATE OR REPLACE TABLE " + table.getName() + " WITH (location = '" + location + "') AS SELECT 'abc' as colA", 1);
+            assertTableType("test_schema", table.getName(), MANAGED_TABLE.name());
+        }
+    }
+
+    @Test
+    public void testCreateOrReplaceTableWithChangeInLocationForManagedTable()
+    {
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "test_create_or_replace_change_location_", " (a BIGINT) ")) {
+            String location = "s3://%s/%s".formatted(bucketName, randomNameSuffix());
+            assertQueryFails(
+                    "CREATE OR REPLACE TABLE " + table.getName() + " (a BIGINT) WITH (location = '%s')".formatted(location),
+                    "The provided location '%s' does not match the existing table location '.*'".formatted(location));
+
+            assertLatestTableOperation(table.getName(), CREATE_TABLE_OPERATION);
+        }
+    }
+
+    @Test
+    public void testCreateOrReplaceTableWithChangeInLocationForExternalTable()
+    {
+        String location = "s3://%s/%s".formatted(bucketName, randomNameSuffix());
+        try (TestTable table = new TestTable(
+                getQueryRunner()::execute,
+                "test_create_or_replace_change_location_",
+                " (a BIGINT) WITH (location = '%s')".formatted(location))) {
+            assertQueryFails(
+                    "CREATE OR REPLACE TABLE " + table.getName() + " (a BIGINT) WITH (location = '%s_2')".formatted(location),
+                    "The provided location '%1$s_2' does not match the existing table location '%1$s'".formatted(location));
+
+            assertLatestTableOperation(table.getName(), CREATE_TABLE_OPERATION);
```

Comment:
maybe verify that the table location has not changed.

Reply:
Added additional assertion.